ec2/roles/jboss-app-deploy/tasks/main.yml

```diff
@@ -0,0 +1,38 @@
+---
+# tasks file for jboss-app-deploy
+
+- name: Create /home/fedora/local directory
+  file:
+    path: /home/fedora/local
+    state: directory
+    owner: fedora
+    group: fedora
+
+- name: Copy the war  to /home/fedora/local
+  ansible.builtin.synchronize:
+    src: "{{ role_path }}/files/{{ app_name }}.war"
+    dest: "/home/fedora/local/"
+  when: warfile_deploy
+
+- name: Deploy the application with Maven to EAP
+  ansible.builtin.copy:
+    src: "/home/fedora/local/{{ app_name }}.war"
+    dest: "{{ jboss_install_home }}/deployments/"
+    remote_src: yes
+    owner: "{{ jboss_owner }}"
+    group: "{{ jboss_group }}"
+    mode: '0644'
+  when: warfile_deploy
+  become: true
+
+- name: Pause play until a URL is reachable from this host
```

Comment:
@fabianvf I changed it to use uri module and added retries. I think its much neater imo. PTAL. Thanks